src/cryptosettings.cpp

```diff
@@ -310,24 +311,29 @@ bool CryptoSettings::writeEncryptedChachaPolyFile(
   }
 
   QByteArray header;
-  QByteArray metadata = getMetaData();
-  if (metadata.isEmpty()) {
+  if (fileVersion == CryptoSettings::EncryptionChachaPolyV1) {
     // Encrypted V1 Header: Just the version.
-    header.append(1, CryptoSettings::EncryptionChachaPolyV1);
-  } else if (metadata.length() <= UINT16_MAX) {
+    header.append(1, fileVersion);
+  } else if (fileVersion == CryptoSettings::EncryptionChachaPolyV2) {
+    QByteArray metadata = getMetaData();
+    if (metadata.length() > UINT16_MAX) {
+      logger.error() << "Failed to write encrypted file header: too long";
+      return false;
+    }
+
     // Encrypted V2 Header:
     // +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
     // | Version | Reserved  | Metadata Length | Metadata  |
     // |  8-bit  |   8-bit   |     16-bit      | variable  |
     // +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
     header.reserve(ENCRYPTED_V2_HEADER_SIZE + metadata.length());
-    header.append(1, CryptoSettings::EncryptionChachaPolyV2);
+    header.append(1, fileVersion);
     header.append(1, 0);
     header.append(1, metadata.length() & 0xff);
     header.append(1, (metadata.length() >> 8) & 0xff);
     header.append(metadata);
-  } else {
-    logger.error() << "Failed to write encrypted file header: too long";
+  } else if (fileVersion != EncryptionChachaPolyV1) {
```

Comment:
We have an early return for ChachaPolyV1. Could this have any bad consequences, especially for non-Linux systems?

Reply:
Uh, good catch, this if/else flow is bit of a mess.

The intention was that we can `writeEncryptedChachaPolyFile` to write an encrypted settings file then call `getPrefferedKey()` to determine which format the platform wants to use to select V1 or V2. If the platform gives us anything other than V1 or V2 then we shouldn't have gotten here or something is wrong with the `switch` statement in `CryptoSettings::writeFile()`.

Reply:
Thanks. Makes sense now.